src/examples/mfem_mortar_lm_patch.cpp

```diff
@@ -0,0 +1,449 @@
+  // performed by calling two methods: 1) registerMfemCouplingScheme() (steps 1
+  // and 2) and 2) updateMfemParallelDecomposition() (steps 3, 4, and 5).
+  // registerMfemCouplingScheme() is called here and
+  // updateMfemParallelDecomposition() is typically called before calling
+  // update().
+  int coupling_scheme_id = 0;
+  int mesh1_id = 0;
+  int mesh2_id = 1;
+  tribol::registerMfemCouplingScheme(
+    coupling_scheme_id, mesh1_id, mesh2_id,
+    *pmesh, coords, mortar_attrs, nonmortar_attrs,
+    tribol::SURFACE_TO_SURFACE, 
+    tribol::NO_SLIDING, 
+    tribol::SINGLE_MORTAR, 
+    tribol::FRICTIONLESS,
+    tribol::LAGRANGE_MULTIPLIER,
+    tribol::BINNING_GRID
+  );
+  // The basic Lagrange multiplier options are set here. For this problem, we
+  // ask Tribol to compute a contact residual and a Jacobian (though we only use
+  // the Jacobian).
+  tribol::setLagrangeMultiplierOptions(
+    coupling_scheme_id,
+    tribol::ImplicitEvalMode::MORTAR_RESIDUAL_JACOBIAN
+  );
+
+  // save initial configuration
+  paraview_datacoll.Save();
+  visit_datacoll.Save();
+  
+  // Update the cycle information for the data collections. Also update time
+  // with a pseudotime for the solution.
+  paraview_datacoll.SetCycle(1);
+  paraview_datacoll.SetTime(1.0);
+  paraview_datacoll.SetTimeStep(1.0);
+  visit_datacoll.SetCycle(1);
+  visit_datacoll.SetTime(1.0);
+  visit_datacoll.SetTimeStep(1.0);
+
+  // This creates the parallel adjacency-based mesh redecomposition. It also
+  // constructs new Tribol meshes as subsets of the redecomposed mesh.
+  tribol::updateMfemParallelDecomposition();
+  double dt {1.0};  // time is arbitrary here (no timesteps)
+  // This API call computes the contact response and Jacobian given the current
+  // mesh configuration.
+  tribol::update(1, 1.0, dt);
+
+  // HypreParMatrices holding the Jacobian from contact and stored in an MFEM
+  // block operator are returned from this API call.
+  auto A_blk = tribol::getMfemBlockJacobian(coupling_scheme_id);
+  // Add the Jacobian from the elasticity bilinear form to the top left block
+  A_blk->SetBlock(0, 0, A.release());
+  timer.stop();
+  SLIC_INFO_ROOT(axom::fmt::format(
+    "Time to setup Tribol and compute Jacobian: {0:f}ms", timer.elapsedTimeInMilliSec()
+  ));
+
+  timer.start();
+  // Create a block RHS vector storing forces and gaps. Note no external forces
+  // are present in this problem.
+  mfem::BlockVector B_blk { A_blk->RowOffsets() };
+  B_blk = 0.0;
+  // Create a block solution vector storing displacement and pressures.
+  mfem::BlockVector X_blk { A_blk->ColOffsets() };
+  X_blk = 0.0;
+
+  // This API call returns the mortar nodal gap vector to an (uninitialized)
+  // vector. The function sizes and initializes the vector.
+  mfem::Vector g;
+  tribol::getMfemGap(coupling_scheme_id, g);
+
+  // Apply a transpose prologation operator on the submesh: maps dofs stored in
+  // dual field g to tdofs stored in G for parallel solution of the linear
```

Comment:
Per my comment about a host-code needing to know about a grid function on the submesh...is this the code that takes 'g' on the submesh to G on the parent mesh? What is G? Maybe make this more explicit if this is the bit of code a host-code would need in order to handle pressures? Let's discuss.

Reply:
This is related to how MFEM works in parallel.  We generally solve things on the "true" degrees of freedom (i.e. not including ghost nodes) and then apply restriction/prolongation operators to build the full on-rank representation (i.e. including ghost nodes).  G is the gap vector on the true degrees of freedom and g is the full gap vector.